services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -320,8 +320,20 @@ protected Response handleSamlRequest(String samlRequest, String relayState) {
         }
 
         protected Response logoutRequest(LogoutRequestType request, String relayState) {
-            String brokerUserId = config.getAlias() + "." + request.getNameID().getValue();
+            if (request.getNameID() == null && request.getBaseID() == null && request.getEncryptedID() == null){
+                logger.error("SAML IdP Logout request must contain at least one of BaseID, NameID and EncryptedID");
+                event.error(Errors.INVALID_SAML_LOGOUT_REQUEST);
+                return ErrorPage.error(session, null, Response.Status.BAD_REQUEST, Messages.IDENTITY_PROVIDER_LOGOUT_FAILURE);
+            }
+
             if (request.getSessionIndex() == null || request.getSessionIndex().isEmpty()) {
+                if (request.getNameID() == null){
+                    //TODO this need to be implemented
```

Comment:
The case of not sending SessionIndex and send either baseId or encryptedId should be implemented.
I do not know how. That's why I throw an error message instead of NullPointer Exception in current Keycloak.

Our SAML IdP use case send SessionIndex with BaseID (or EncryptedID). This leads to NullPointer exception. So my PR fix it.

Reply:
Do you mind rebasing your PR. Other than that, LGTM.